torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2704,7 +2704,7 @@ def create_symbolic_sizes_strides_storage_offset(
         # we may have an unnessary shape speciliazation for y.
         def maybe_specialize_sym_int_with_hint(maybe_sym) -> int:
             assert isinstance(maybe_sym, (int, torch.SymInt))
```

Comment:
Couldn't this assert be updated too? If is_symbolic has a typing_extensions.TypeGuard annotation at least.

Reply:
Using is_symbolic would be testing something different than what we want here. This assert is saying that we are okay with either ints or SymInts whether or not those SymInts are symbolic.